process/transactionProcessor.go

```diff
@@ -216,7 +216,14 @@ func (tp *TransactionProcessor) TransactionCostRequest(tx *data.Transaction) (st
 
 // GetTransaction should return a transaction from observer
 func (tp *TransactionProcessor) GetTransaction(txHash string) (*data.FullTransaction, error) {
-	return tp.getTxFromObservers(txHash)
+	tx, err := tp.getTxFromObservers(txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	tx.HyperblockNonce = tx.NotarizedAtDestinationInMetaNonce
```

Comment:
if the data is already there why do you need it in duplicated fields ?

Reply:
These fields are for advanced usage that is aware of sharding:

```
NotarizedAtSourceInMetaNonce
NotarizedAtSourceInMetaHash
NotarizedAtDestinationInMetaNonce
NotarizedAtDestinationInMetaHash
```
While these are for sharding-agnostic scenarios (e.g. exchanges):

```
HyperblockNonce
HyperblockHash
```

Indeed, there's a duplication because:

```
HyperblockNonce = NotarizedAtDestinationInMetaNonce
HyperblockHash = NotarizedAtDestinationInMetaHash
```

Can we keep the redundant fields like this (at least for the moment)?

Reply:
for the moment we might keep it.